test/tchannel_bridge.js

```diff
@@ -0,0 +1,125 @@
+    ];
+
+    _.each(options, (o) => {
+
+        it (o.description + ' spans propagate through tchannel and preserve parent span properties', (done) => {
+            let originalSpan = tracer.startSpan('futurama');
+            originalSpan.setBaggageItem('leela', 'fry');
+            let contextForOutgoingCall = { span: originalSpan };
+
+            let server = new TChannel({
+                serviceName: 'server',
+                timeout: BIG_TIMEOUT,
+                // force tracing on  in order to prove that overriding works
+                trace: true,
+                forceTrace: true
+            });
+            // Server calls client channel after it starts listening.
+            server.listen(4040, '127.0.0.1', onServerListening);
+
+            // Create the toplevel client channel.
+            let client = new TChannel({
+                // force tracing on  in order to prove that overriding works
+                trace: true,
+                forceTrace: true
+            });
+
+            // Create the client subchannel that makes requests.
+            let clientSubChannel = client.makeSubChannel({
+                serviceName: 'server',
+                peers: ['127.0.0.1:4040']
+            });
+
+            // Wrap the subchannel in an encoding
+            let encodedChannel = o.channelEncoding({
+                channel: clientSubChannel,
+                entryPoint: path.join(__dirname, 'thrift', 'echo.thrift') // ignored in json case
+            });
+
+            // register the server request function.
+            let context = {};
+            encodedChannel.register(server, 'Echo::echo', context, bridge.tracedHandler(handleServerReq));
+            function handleServerReq(context, req, head, body, callback) {
+                // assert that the serverSpan is a child of the original span
+                assert.equal(originalSpan.context().traceIdStr, context.span.context().traceIdStr);
+                assert.equal(originalSpan.context().spanIdStr, context.span.context().parentIdStr);
+                server.close();
+                client.close();
+            }
+
+            function onServerListening() {
+                // Outgoing tchannel call is traced
+                let tracedChannel = bridge.tracedChannel(encodedChannel, contextForOutgoingCall);
+                let req = tracedChannel.request({
+                    serviceName: 'server',
+                    headers: { cn: 'echo' },
+                    timeout: BIG_TIMEOUT
+                });
+
+                // headers are left empty in order to prove that tracedChannel populates them
+                let emptyHeaders = {};
+                req.send('Echo::echo', emptyHeaders, { value: 'some-string' }, (err, res, headers, body) => {
+                    assert.equal(reporter.spans.length, 2);
+                    // the first span to be reported is the server span
```

Comment:
I am not 100% if async nature of Node would guarantee that. In tchannel-go I certainly saw these being flipped.